app.js

```diff
@@ -88,3 +91,34 @@ function installDropHandler (archive) {
     }
   })
 }
+
+function attachSpeedometer (archive) {
+  var speed = speedometer(1)
+  var $els = {
+    speed: document.getElementById('speed'),
+    upload: document.getElementById('upload-speed'),
+    download: document.getElementById('download-speed')
+  }
+  var timer;
+  function update (direction, data) {
+    if (!data.length) return
+    var bytesPerSecond = speed(data.length)
```

Comment:
is this using the same speedometer for both upload / download? you need one for each :)